src/zenml/integrations/gcp/service_connectors/gcp_service_connector.py

```diff
@@ -146,6 +168,24 @@ class GCPServiceAccountCredentials(AuthenticationConfig):
         "distribute the service account key JSON to clients instead.",
     )
 
+    @root_validator(pre=True)
+    def validate_service_account_dict(
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Convert the service account credentials to JSON if given in dict format.
+
+        Args:
+            values: The configuration values.
+
+        Returns:
+            The validated configuration values.
+        """
+        if isinstance(values.get("service_account_json"), dict):
```

Comment:
This is just something left over from the "use service connectors to authenticate secrets stores" PR and revealed during testing this PR.